src/geotop/input.cc

```diff
@@ -710,20 +699,30 @@ used default values" << std::endl;
     // Completing of "land" (of the type LAND):
 
     // Initialize matrix of shadow
-    land->shadow=new_shortmatrix(Nr,Nc);
-    initialize_shortmatrix(land->shadow, 0); /* initialized as if it was always NOT in shadow */
+    land->shadow.reset(new Matrix<short>{Nr,Nc}); /* initialized as if it was always NOT in shadow */
 
     // Check that there aren't cell with an undefined land use value
     z = 0.;
     l = 0;
+
     do
     {
         l++;
         z += sl->pa->co[1][jdz][l];
     }
     while (l<Nl && z < z_transp);
-    land->root_fraction=new_doublematrix(par->n_landuses, l);
-    initialize_doublematrix(land->root_fraction, 0.0);
+
+    // We increase l and z_transp to avoid segmentation fault when accessing (*land->root_fraction)(lu,Nl)
```

Comment:
it was one of the solutions to avoid segfault for  WG1_2.0_001 test case. I'll discuss with you the best solution but since I chose another option these comments wll be removed